### DISCUSSION TITLE
How to debug high CPU load?

### DISCUSSION THREAD

**Discussion by mifritscher** [arendst/Tasmota#20056] [Q&A]
My sensor says this:

```
08:02:55.446 MQT: tele/tasmota_324E10/STATE = {"Time":"2023-11-19T08:02:55","Uptime":"0T14:28:12","UptimeSec":52092,"Heap":98,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":2131,"MqttCount":10,"Berry":{"HeapUsed":3,"Objects":33},"POWER1":"ON","POWER2":"ON","POWER3":"ON","Wifi":{"AP":1,"SSId":"Test","BSSId":"44:4E:6D:25:FF:FF","Channel":1,"Mode":"11n","RSSI":86,"Signal":-57,"LinkCount":5,"Downtime":"0T00:03:49"}}
08:02:55.515 MQT: tele/tasmota_324E10/SENSOR = {"Time":"2023-11-19T08:02:55","Switch1":"ON","Switch2":"ON","DS18B20":{"Id":"346AE8086461","Temperature":21.4},"BME680":{"Temperature":20.6,"Humidity":47.8,"DewPoint":9.1,"Pressure":963.8,"Gas":73.51},"VEML7700":{"Illuminance":0,"WhiteContent":0},"HC8":{"Model":"HC8","CarbonDioxide":"1115"},"ESP32":{"Temperature":80.0,"HallEffect":3},"PressureUnit":"hPa","TempUnit":"C"}
```

No berry-scripts, rules are disabled for this test, MQTT is running.

Other config:
```
Setoption114 1
SwitchMode1 1
SwitchMode2 1
#SwitchTopic 0
#Setoption114 1
Rule1 on Switch1#state=1 do publish stat/%topic%/PIR1 ON endon on Switch1#state=0 do Publish stat/%topic%/PIR1 OFF endon
Rule1 1
Rule2 on Switch2#state=1 do publish stat/%topic%/SND1 ON endon on Switch1#state=0 do Publish stat/%topic%/SND1 OFF endon
Rule2 1

TelePeriod 30

Setoption146 1
```

Problem: `Sleep":50,"LoadAvg":2131`  - here the LoadAvg seems to be quite high? Any hints how to debug that?

I'm running tasmota 13.2.0.2.

**Comment by mifritscher** [comment]
Ok, seems to be the "Buzzer" thing (  https://github.com/arendst/Tasmota/discussions/11969 ). But the temperature stays high even if disabling the Buzzer (and then the load is normal)...

**Reply by sfromis** [reply]
Do not assume that "LoadAvg" means CPU load, and do not assume that "ESP32 Temperature" is actual temperature. Seriously :grimacing: 
The default for `SetOption146` is "off", to avoid the understandable confusion.

**Reply by mifritscher** [reply]
Hmm, ok, is the sensor _that_ off? Because the 3 in use all says about 60-80 °C, while a "fresh" tasmeta one says 30°C. And the ones in use are indeed quite warm (finger and measurement - about 35°C), while the fresh one stays cool.

Is there a way to "underclock" the ESP32 with tasmota, e.g. to the 80 Mhz the ESP 8266 uses? The ESP32 itself has this capability. I indeed use a single core ESP 8266 in similiar scenarios indeed without any problems. Ok, too less pins^^

**Reply by sfromis** [reply]
I've seen "ESP32 Temperature" show as around 10 degrees on a day with normal indoor temperature of around 24, or much higher even right after power on. I'm not trusting it for anything. If I wanted to track power consumption, I'd measure this instead.

Using the command `CpuFrequency`, you can select 80/160/240 MHz.

**Reply by sfromis** [reply]
If your real temperature sensor is being heated, you have a bad build which you should fix. In addition to distance, also look at heat-conducting surfaces like a PCB (cutting open lines in the PCB may help), and airflow, where vents and low sensor placement helps. You are right that an offset is no solution, as you cannot expect the deviation to be constant.
